### PR TITLE
chore: alphabetize template list

### LINKED ISSUE #2836: Template lists should be alphabetical
## Problem
Both in the web UI and the CLI.  Right now they're ordered by date of creation, which isn't helpful, and in the web UI, you can't even tell that's what's happening because it only has an updated column not a created one.

## Definition of Done
Template list should be auto sorted alphabetically

### PR DESCRIPTION
**Pull request by bpmct**
fixes #2836

### CODE REVIEW

**Review by kylecarbs — APPROVED** [review]
LGTM

File: coderd/database/queries/templates.sql [hunk 1]
```diff
@@ -51,7 +51,7 @@ LIMIT
 
 -- name: GetTemplates :many
 SELECT * FROM templates
-ORDER BY (created_at, id) ASC
+ORDER BY (name, id) ASC
```
Comment (mafredri) [inline coderd/database/queries/templates.sql:54]: This change needs to be reflected in the test as well as the databasefake.

**Review by mafredri — APPROVED** [review]
A sensible change considering the client can't request different ordering.

Test failures need addressing but otherwise good to go. 👍🏻

**Review by dwahler — APPROVED** [review]
LGTM!

**Review by kylecarbs — APPROVED** [review]

**Review by mafredri — APPROVED** [review]
Nice!

**Pull request merged**